library/proc_macro/src/lib.rs

```diff
@@ -815,6 +815,16 @@ pub enum Delimiter {
     /// "macro variable" `$var`. It is important to preserve operator priorities in cases like
     /// `$var * 3` where `$var` is `1 + 2`.
     /// Invisible delimiters might not survive roundtrip of a token stream through a string.
+    ///
+    /// <div class="warning">
+    ///
+    /// Note: rustc currently (1.77 and below) ignores the grouping of tokens
+    /// delimited by `None` in the output of a proc_macro. Thus it will not
+    /// preserve operator priorities as indicated above. The other `Delimiter`
+    /// variants should be used instead in this context. For details, see
+    /// [rust-lang/rust#67062](https://github.com/rust-lang/rust/issues/67062).
+    ///
```

Comment:
The situation is slightly more complicated than described here.  In particular, `None`-delimited groups that appear in `Debug` output from `TokenStream`s *provided to* a proc macro, which *came from* a macro_rules matcher, *are* effective!  But only if the proc_macro doesn't disassemble and reassemble things "too much".  See the example in https://github.com/rust-lang/rust/issues/124974

Maybe:
```suggestion
    ///
    /// `TokenStream`s which contain tokens pre-processed by `macro_rules!` matchers may
    /// seem to contain `None`-delimited groups, when viewed by a proc macro.
    /// The representation seen by a proc macro, in these circumstances, is not faithful;
    /// but, the underlying (working) representaton is sometimes preserved
    /// if the very same `TokenTree`s are returned by the proc macro.
    ///
```

or something.

Reply:
They do indeed work _potentially_. But with the current situation it's really hard to give a guarantee about it, except for stating that they "might" work, which is fairly useless for everyone involved. If we put this in here there should be a good way to judge if it actually applies to a certain use case.

Of course ideally the original bug will just get fixed.

Reply:
I think it's better to say "might" than to gaslight the poor programmer with a thing that sometimes does one thing and sometimes another, in apparently identical circumstances.

But please don't let me stand in the way of this being documented *at all*!